### DISCUSSION TITLE
using external js libs

### DISCUSSION THREAD

**Discussion by dncpax** [Budibase/budibase#2517] [Help] [answered]
Hi all. Just wondering it there's anyway to use external libs like openalyers.org? Either in writing js in the builder or creating custom components/widgets/controls...

Ok, I see there's an embed component. How does it interact with other controls? For instance, select a value on a dropdown, and then make a map zoomin into a location?

Thanks

**Comment by shogunpurple** [comment, marked as answer]
Hi @dncpax, this is not yet possible with Budibase, but we do have plans for much more advanced JS integration over the coming months.

**Reply by tmdoit** [reply]
Hi @shogunpurple!
Can you say something more about "more advanced JS integration"?
I'm also interested in using external libraries, for now I have requirement to use "jsrsasign-js" lib, in general for JWT generation which I will use to query ext REST API.

**Reply by ellemedit** [reply]
@shogunpurple Hi, this is most important feature for my company project. Is there plan to allow contribution for this?

**Comment by daawaan4x** [comment]
Hi @dncpax, if you're fine with building Budibase from source, you might want to check my pull request for a better map component in #5342. I'm also currently using Budibase for a GIS project
